### PR TITLE
billing: show legacy plan pricing notice

### PR DESCRIPTION
**Pull request by elie222**
# User description
Adds a small safeguard for subscriptions that are still on legacy Stripe prices.

It exposes the active Stripe price ID to the client and detects when a plan is using grandfathered pricing. It shows a legacy pricing notice in billing settings and on the premium page. It also adds a focused unit test for the price check helper.

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
Expose <code>stripePriceId</code> through the <code>user/me</code> API so billing flows can determine when a subscription still uses grandfathered Stripe pricing, and extend the premium configuration helpers to detect those legacy prices. Surface that status in the billing settings and premium pricing UI so customers understand they are on legacy rates while ensuring the dependency lockfile stays aligned with the CLI packaging expectations.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1927?tool=ast&topic=Legacy+notice+UI>Legacy notice UI</a>
        </td><td>Display the detected legacy status in the billing settings and premium pricing pages by reusing <code>getPremiumTierName</code>, showing contextual text via <code>AlertBasic</code>, and keeping the account-level plan summary up to date so customers see consistent messaging about grandfathered rates.<details><summary>Modified files (2)</summary><ul><li>apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx</li>
<li>apps/web/app/(app)/premium/Pricing.tsx</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>analytics-fix-Stripe-t...</td><td>March 12, 2026</td></tr>
<tr><td>josh@jshwrnr.com</td><td>Globalize-settings-rou...</td><td>February 10, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1927?tool=ast&topic=Legacy+pricing+logic>Legacy pricing logic</a>
        </td><td>Implement legacy Stripe detection by adding <code>hasLegacyStripePriceId</code>/<code>shouldShowLegacyStripePricingNotice</code> helpers in the premium config, expose <code>stripePriceId</code> in <code>getUser</code>, and cover the logic with targeted unit tests so other flows can reliably know when a plan uses grandfathered pricing.<details><summary>Modified files (3)</summary><ul><li>apps/web/app/(app)/premium/config.test.ts</li>
<li>apps/web/app/(app)/premium/config.ts</li>
<li>apps/web/app/api/user/me/route.ts</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>pricing</td><td>March 10, 2026</td></tr>
<tr><td>josh@jshwrnr.com</td><td>Backend-fixes-for-mobi...</td><td>March 09, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1927?tool=ast&topic=Lockfile+cleanup>Lockfile cleanup</a>
        </td><td>Clean up the <code>pnpm-lock.yaml</code> entry by removing the unused <code>packages/api-cli</code> dependency block to reflect the current dependency tree.<details><summary>Modified files (1)</summary><ul><li>pnpm-lock.yaml</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>API-add-npm-CLI-for-au...</td><td>March 15, 2026</td></tr>
<tr><td>msoukhomlinov@users.no...</td><td>Add-OpenAI-compatible-...</td><td>February 22, 2026</td></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1927?tool=ast>(Baz)</a>.

### CODE REVIEW

File: apps/web/app/(app)/premium/config.ts [hunk 1]
```diff
@@ -133,6 +133,44 @@ export function getStripePriceId({
...
+  priceId,
+}: {
+  tier: PremiumTier | null | undefined;
+  priceId: string | null | undefined;
+}): boolean {
+  if (!tier || !priceId) return false;
+
+  const currentPriceId = getStripePriceId({ tier });
+  if (!currentPriceId) return false;
+
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/premium/config.ts:147]: Updated this to fall back to `getStripeSubscriptionTier({ priceId })` when `tier` is missing, so legacy pricing still gets detected for null-tier subscriptions. I also added test coverage for both current and legacy price IDs in the null-tier path.

File: apps/web/app/(app)/premium/Pricing.tsx [hunk 2]
```diff
@@ -44,11 +49,17 @@ export default function Pricing(props: PricingProps) {
...
+    ? "welcome_upgrade"
+    : "app_premium";
   const displayedTiers = props.displayTiers || tiers;
   const hasExistingSubscription = Boolean(
     premium?.stripeSubscriptionId || premium?.lemonSqueezyCustomerId,
   );
+  const isLegacyStripePlan = hasLegacyStripePriceId({
+    tier: premium?.tier,
+    priceId: premium?.stripePriceId,
+  });
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/premium/Pricing.tsx]: Valid. I gated the notice so it only runs for Stripe-backed subscriptions without a Lemon customer record, which avoids false legacy warnings on migrated Lemon accounts.

File: apps/web/app/(app)/premium/config.ts [hunk 3]
```diff
@@ -133,6 +133,44 @@ export function getStripePriceId({
...
+  return currentPriceId !== priceId;
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/premium/config.ts]: Valid. I tightened `hasLegacyStripePriceId` to only return true for configured `oldPriceIds`, while still deriving the tier from `priceId` when `tier` is missing. I also added a regression test for unknown non-current price IDs.

File: apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx [hunk 4]
```diff
@@ -12,12 +12,20 @@ import {
...
+  getPremiumTierName,
+  hasLegacyStripePriceId,
+} from "@/app/(app)/premium/config";
 
 export function BillingSection() {
   const { premium, isPremium, isLoading } = usePremium();
+  const isLegacyStripePlan = hasLegacyStripePriceId({
+    tier: premium?.tier,
+    priceId: premium?.stripePriceId,
+  });
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx]: Valid. I applied the same provider guard in billing settings, so the legacy pricing notice only shows for Stripe-backed subscriptions and not mixed Stripe/Lemon records.

File: apps/web/app/(app)/premium/Pricing.tsx [hunk 5]
```diff
@@ -44,11 +49,20 @@ export default function Pricing(props: PricingProps) {
...
   const hasExistingSubscription = Boolean(
     premium?.stripeSubscriptionId || premium?.lemonSqueezyCustomerId,
   );
+  const isLegacyStripePlan =
+    !!premium?.stripeSubscriptionId &&
+    !premium?.lemonSqueezyCustomerId &&
+    hasLegacyStripePriceId({
+      tier: premium?.tier,
+      priceId: premium?.stripePriceId,
+    });
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/premium/Pricing.tsx]: Updated this to use a shared `shouldShowLegacyStripePricingNotice(premium)` helper instead of the old inline guard. The helper now keys off an active or trialing Stripe subscription status plus the legacy Stripe price check, so migrated records with both provider IDs are no longer suppressed just because a Lemon customer ID exists.

File: apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx [hunk 6]
```diff
@@ -12,12 +12,23 @@ import {
...
+  hasLegacyStripePriceId,
+} from "@/app/(app)/premium/config";
 
 export function BillingSection() {
   const { premium, isPremium, isLoading } = usePremium();
+  const isLegacyStripePlan =
+    !!premium?.stripeSubscriptionId &&
+    !premium?.lemonSqueezyCustomerId &&
+    hasLegacyStripePriceId({
+      tier: premium?.tier,
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx]: Extracted the predicate into `shouldShowLegacyStripePricingNotice(premium)` in `app/(app)/premium/config.ts` and switched both BillingSection and Pricing to call it. The shared helper is covered in `config.test.ts` so the rule only lives in one place now.

File: apps/web/app/(app)/premium/Pricing.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/premium/Pricing.tsx]: Addressed in the shared helper. The notice now requires an active or trialing Stripe subscription status, so canceled past Stripe subscriptions no longer show the grandfathered pricing message.

File: apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx [hunk 7]
```diff
@@ -12,12 +12,23 @@ import {
...
+      priceId: premium?.stripePriceId,
+    });
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx]: Addressed by the shared helper as well. The notice no longer depends on `stripeSubscriptionId`; it uses the Stripe subscription status plus the legacy price check, which covers premium users whose Stripe ID was not persisted but whose Stripe status is still present.

File: apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx [hunk 8]
```diff
@@ -12,12 +12,23 @@ import {
...
+import {
+  getPremiumTierName,
+  hasLegacyStripePriceId,
+} from "@/app/(app)/premium/config";
 
 export function BillingSection() {
   const { premium, isPremium, isLoading } = usePremium();
+  const isLegacyStripePlan =
+    !!premium?.stripeSubscriptionId &&
+    !premium?.lemonSqueezyCustomerId &&
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/[emailAccountId]/settings/BillingSection.tsx]: Addressed. The Lemon customer guard was removed from the notice predicate, and migrated accounts with both provider IDs now rely on the active Stripe status plus `stripePriceId` to decide whether the legacy Stripe notice should appear.

File: apps/web/app/(app)/premium/config.ts [hunk 9]
```diff
@@ -133,6 +133,47 @@ export function getStripePriceId({
...
+  priceId,
+}: {
+  tier: PremiumTier | null | undefined;
+  priceId: string | null | undefined;
+}): boolean {
+  if (!priceId) return false;
+
+  const resolvedTier = tier || getStripeSubscriptionTier({ priceId });
+  if (!resolvedTier) return false;
+
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/premium/config.ts:147]: I did not implement this in the UI helper. Without a persisted `stripePriceId`, the client does not have a reliable signal to distinguish a current Stripe tier from a legacy Stripe tier, so any fallback here would be guesswork. The right fix would be a separate server-side backfill or Stripe resync path for missing `stripePriceId` records.

File: apps/web/app/(app)/premium/config.ts [hunk 10]
```diff
@@ -133,6 +133,47 @@ export function getStripePriceId({
...
+  return (
+    STRIPE_PRICE_ID_CONFIG[resolvedTier]?.oldPriceIds?.includes(priceId) ??
+    false
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/premium/config.ts:150]: I did not change this behavior in this PR. We only treat explicitly configured legacy Stripe price IDs as grandfathered, because falling back to “any non-current price” would reintroduce false positives for unknown or stale price IDs. If we want BASIC/PRO legacy notices as well, we need the authoritative historical Stripe price IDs added to `STRIPE_PRICE_ID_CONFIG` in a follow-up.

File: packages/api/package.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@inbox-zero/api",
-  "version": "0.1.0",
+  "version": "2.29.2",
```
[the remark this answers is not in the document]
Comment (elie222) [reply packages/api/package.json:3]: This is no longer part of the PR diff. I rebased the branch onto current `origin/main`, which removed the unrelated API packaging changes from this pricing PR.
